Vue Component.sublime-syntax

```diff
@@ -493,71 +449,170 @@ contexts:
+                    1: comment.block.html punctuation.definition.comment.begin.html
+                  pop: 1
+                  embed: scope:text.slm
+                  escape: '{{template_content_end}}'
+            - include: template-common
         - tag-generic-attribute-meta
         - tag-generic-attribute-value
     - match: (?=\S)
       set:
         - template-mustache
         - tag-generic-attribute-meta
         - tag-generic-attribute-value
+
+###[ TAG ATTRIBUTES ]#########################################################
+
+  tag-attributes:
+    - meta_prepend: true
+    - include: vue-directive
+
+  tag-class-attribute-value:
+    # for compatibility with ST builds before 4114
+    - meta_prepend: true
+    - meta_include_prototype: false
+
+  tag-generic-attribute-value:
+    # for compatibility with ST builds before 4114
+    - meta_prepend: true
+    - meta_include_prototype: false
+
+  tag-href-attribute-value:
+    # for compatibility with ST builds before 4114
+    - meta_prepend: true
+    - meta_include_prototype: false
+
+  tag-id-attribute-value:
+    # for compatibility with ST builds before 4114
+    - meta_prepend: true
+    - meta_include_prototype: false
+
+  tag-attribute-value-content:
+    - meta_prepend: true
+    - include: mustache-interpolations
+
+  strings-common-content:
+    - meta_prepend: true
+    - include: mustache-interpolations
+
+###[ MUSTAGE TEMPLATES ]######################################################
+
+  mustache-interpolations:
+    - match: (?={{)
+      push: mustache-interpolation-content
+
+  mustache-interpolation-content:
+    - clear_scopes: 1
+    - meta_scope: meta.interpolation.html
+    - include: mustache-templates
+    - include: immediately-pop
+
+  mustache-templates:
+    - match: '{{'
+      scope: meta.template.vue punctuation.definition.template.begin.html
+      embed: scope:source.js
+      embed_scope: meta.template.vue source.js.embedded.vue
+      escape: '}}'
+      escape_captures:
+        0: meta.template.vue punctuation.definition.template.end.html
+
+###[ VUE DIRECTIVES ]#########################################################
+
+  vue-directive:
+    - match: (?=v-)
+      push:
+        - vue-directive-meta
+        - vue-directive-assignment
+        - tag-generic-attribute-name
+    - match: (?::|@)
+      scope: punctuation.definition.attribute.html
+      push:
+        - vue-directive-meta
+        - vue-directive-assignment
+        - tag-generic-attribute-name
+
+  vue-directive-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.attribute-with-value.directive.html
+    - include: immediately-pop
+
+  vue-directive-assignment:
+    - match: =
+      scope: punctuation.separator.key-value.html
+      set: vue-directive-value
+    - include: else-pop
+
+  vue-directive-value:
+    - match: \"
+      scope: meta.string.html string.quoted.double.html punctuation.definition.string.begin.html
+      pop: 1
+      embed: scope:source.js
```

Comment:
If this embedded the JS syntax's `expression-statement` context, it would highlight mappings such as for class specifications correctly instead of matching them as blocks (and any keys within as labels).

Not a new issue, though.

Reply:
Something like `<p v-attrib="let i = 0">` is not supported then? The snippet is taken from tests.

Reply:
Uhm, I've never seen that, but I can verify whether that does anything or errors tomorrow. What you usually see here are:

1. expressions / references to values or functions defined in the `<script>` part, e.g. `v-model="variableName"` or `:enabled="true"` or `v-on:hover="functionName"` or `v-if="isEnabled"`
2. function *calls* with special values, such as `@click="onClick($event)` (are evaluated/substituted when called).
3. a special for syntax for `v-for="(item, idx) in items"` (also for-of)